### PR TITLE
fix: improve the logic of handlers to cover scenarios where resources already exist

### PR DESCRIPTION
**Pull request by fabriziosestito**
improve the logic of handlers to cover scenarios where resources already exist

- `CreateCatalogHandler`: skip creation if an `Image` already exists, delete obsolete images.
- `GenerateSBOMHandler`: skip creation if a `SBOM` already exists
- `ScanSBOMHandler`: create or update `VulnerabilityReport`

### CODE REVIEW

File: internal/handlers/create_catalog.go [hunk 1]
```diff
@@ -75,7 +76,7 @@ func (h *CreateCatalogHandler) Handle(message messaging.Message) error {
 		return fmt.Errorf("cannot discover repositories: %w", err)
 	}
 
-	var imageNames []string
+	newImageNames := sets.Set[string]{}
```
Comment (flavio) [inline internal/handlers/create_catalog.go:79]: nit: `newImageNames` makes me think about the "net new image", the one that have been discovered and are not yet known (as in, do not have an Image CR associated).

Can we pick a different name?

File: internal/handlers/create_catalog.go [hunk 2]
```diff
@@ -259,6 +264,28 @@ func (h *CreateCatalogHandler) transportFromRegistry(registry *v1alpha1.Registry
 	return transport
 }
 
+// deleteObsoleteImages deletes images that are not present in the discovered registry anymore.
+func (h *CreateCatalogHandler) deleteObsoleteImages(ctx context.Context, existingImageNames sets.Set[string], newImagesNames sets.Set[string], namespace string) error {
+	for existingImageName := range existingImageNames {
+		if newImagesNames.Has(existingImageName) {
+			continue
+		}
```
Comment (flavio) [inline internal/handlers/create_catalog.go:272]: nit: isn't there an "intersection"-like method provided by this `Set` struct?

**Review by flavio — COMMENTED** [review]
LGTM, I left two minor comments

File: internal/handlers/create_catalog.go (hunk 2 above)
Comment (fabriziosestito) [reply internal/handlers/create_catalog.go:272]: We could use `sets.Difference` but it doesn't change much as we need to cycle to delete the images anyway. Also, this way we will cycle twice.

**Pull request merged**